swift-recon.py

```diff
@@ -351,9 +348,18 @@ def main():
             maas_common.status_err('no ring provided to check')
         stats = swift_replication(args.ring)
     else:
-        maas_common.status_err(
-            'unrecognized command "{0}"'.format(args.recon)
-            )
+        raise TypeError('unrecognized command "{0}"'.format(args.recon))
+    return stats
+
+
+def main():
+    parser = make_parser()
+    args = parser.parse_args()
+
+    try:
+        stats = get_stats_from(args)
+    except (ValueError, TypeError) as e:
```

Comment:
I'll happily make all the wonky custom exceptions we could possibly want, but I'm rather convinced we shouldn't be mixing our parsing/data-fetching functions with printing data. Having all of the cases where we print data in one place makes everything a lot cleaner in the end.